test/Common/HttpApiHandlerTests.cs

```diff
@@ -913,6 +913,75 @@ public async Task GetEntity_Returns_State_Or_HTTP_404(bool hasKey, bool exists)
+
+            var testRequest = new HttpRequestMessage
+            {
+                Method = HttpMethod.Get,
+                RequestUri = uriBuilder.Uri,
+            };
+
+            var list = (IReadOnlyCollection<DurableEntityStatus>)new List<DurableEntityStatus>
+            {
+                new DurableEntityStatus
+                {
+                    EntityId = new EntityId("DoThis", "one"),
+                    LastOperationTime = new DateTime(2018, 3, 10, 10, 10, 10, DateTimeKind.Utc),
+                    State = null,
+                },
+                new DurableEntityStatus
+                {
+                    EntityId = new EntityId("DoThat", "two"),
+                    LastOperationTime = new DateTime(2018, 3, 10, 10, 6, 10, DateTimeKind.Utc),
+                    State = null,
+                },
+            };
+
+            var result = new EntityQueryResult() { Entities = list, ContinuationToken = null };
+            var clientMock = new Mock<IDurableClient>(MockBehavior.Strict);
+
+            clientMock
+                    .Setup(x => x.ListEntitiesAsync(It.IsAny<EntityQuery>(), It.IsAny<CancellationToken>()))
+                    .Returns(Task.FromResult(result));
+
+            var httpApiHandler = new ExtendedHttpApiHandler(clientMock.Object);
+
+            var lastOperationTimeFrom = new DateTime(2018, 3, 10, 10, 1, 0, DateTimeKind.Utc);
+            var lastOperationTimeTo = new DateTime(2018, 3, 10, 10, 23, 59, DateTimeKind.Utc);
+
+            var getStatusRequestUriBuilder = new UriBuilder(TestConstants.NotificationUrl);
+            getStatusRequestUriBuilder.Path += $"/Entities/";
+            getStatusRequestUriBuilder.Query = $"lastOperationTimeFrom={WebUtility.UrlEncode(lastOperationTimeFrom.ToString())}&lastOperationTimeTo={WebUtility.UrlEncode(lastOperationTimeTo.ToString())}&fetchInput=false&top=100";
+
+            var requestMessage = new HttpRequestMessage
+            {
+                Method = HttpMethod.Get,
+                RequestUri = getStatusRequestUriBuilder.Uri,
+            };
+            requestMessage.Headers.Add("x-ms-continuation-token", "XXXX-XXXXXXXX-XXXXXXXXXXXX");
+
+            var responseMessage = await httpApiHandler.HandleRequestAsync(requestMessage);
+            Assert.Equal(HttpStatusCode.OK, responseMessage.StatusCode);
+            Assert.Equal("YYYY-YYYYYYYY-YYYYYYYYYYYY", responseMessage.Headers.GetValues("x-ms-continuation-token").FirstOrDefault());
+            var actual = JsonConvert.DeserializeObject<IList<DurableEntityStatus>>(await responseMessage.Content.ReadAsStringAsync());
+            clientMock.Verify(x => x.ListEntitiesAsync(It.IsAny<EntityQuery>(), It.IsAny<CancellationToken>()));
+            Assert.Equal("DoThis", actual[0].EntityId.EntityName);
+            Assert.Equal("one", actual[0].EntityId.EntityKey);
+            Assert.Equal("DoThat", actual[1].EntityId.EntityName);
+            Assert.Equal("02", actual[1].EntityId.EntityKey);

```

Comment:
Is this test passing? There are a couple things that I would expect wouldn't work, but maybe I'm misunderstanding:
* It's not clear to me why Line 976 would see an `x-ms-continuation-token` that returns `YYYY-...`.
* I would expect in the case where you're querying for a specific entity (`hasName = true`) that only one comes back. This test always seems to assert that both come back.

Reply:
You're right about the YYYY thing, I left that in from another test accidentally.

This is just a unit test to make sure that ListEntitiesAsync is being called, DoThis/DoThat should always be returned.

Reply:
> This is just a unit test to make sure that ListEntitiesAsync is being called, DoThis/DoThat should always be returned.

In that case, I suggest you remove the `hasName` and `entityName` variables from this test code. The URL you are generating does not match the behavior of your mocked HTTP API, which is confusing and may mislead people.